modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/roundstartslime.dm

```diff
@@ -33,212 +51,433 @@
+		alterer,
+		"What do you want the part to become?",
+		"Body Part Alterations",
+		choice_list,
+	)
+	if(!chosen_name_key)
+		return
+
+	var/datum/sprite_accessory/selected_sprite_accessory = GLOB.sprite_accessories[chosen_key][chosen_name_key]
+	alterer.mutant_renderkey = "" //Just in case
+	if(!selected_sprite_accessory.factual)
+		if(selected_sprite_accessory.organ_type)
+			var/obj/item/organ/organ_path = selected_sprite_accessory.organ_type
+			var/slot = initial(organ_path.slot)
+			var/obj/item/organ/got_organ = alterer.getorganslot(slot)
+			if(got_organ)
+				got_organ.Remove(alterer)
+				qdel(got_organ)
+		else
+			alterer.dna.species.mutant_bodyparts -= chosen_key
+	else
+		if(selected_sprite_accessory.organ_type)
+			var/obj/item/organ/organ_path = selected_sprite_accessory.organ_type
+			var/slot = initial(organ_path.slot)
+			var/obj/item/organ/got_organ = alterer.getorganslot(slot)
+			if(got_organ)
+				got_organ.Remove(alterer)
+				qdel(got_organ)
+			organ_path = new selected_sprite_accessory.organ_type
+			var/list/new_acc_list = list()
+			new_acc_list[MUTANT_INDEX_NAME] = selected_sprite_accessory.name
+			new_acc_list[MUTANT_INDEX_COLOR_LIST] = selected_sprite_accessory.get_default_color(alterer.dna.features, alterer.dna.species)
+			alterer.dna.mutant_bodyparts[chosen_key] = new_acc_list.Copy()
+			if(ROBOTIC_DNA_ORGANS in alterer.dna.species.species_traits)
+				organ_path.status = ORGAN_ROBOTIC
+				organ_path.organ_flags |= ORGAN_SYNTHETIC
+			organ_path.build_from_dna(alterer.dna, chosen_key)
+			organ_path.Insert(alterer, 0, FALSE)
+
+		else
+			var/list/new_acc_list = list()
+			new_acc_list[MUTANT_INDEX_NAME] = selected_sprite_accessory.name
+			new_acc_list[MUTANT_INDEX_COLOR_LIST] = selected_sprite_accessory.get_default_color(alterer.dna.features, alterer.dna.species)
+			alterer.dna.species.mutant_bodyparts[chosen_key] = new_acc_list
+			alterer.dna.mutant_bodyparts[chosen_key] = new_acc_list.Copy()
+		alterer.dna.update_uf_block(GLOB.dna_mutant_bodypart_blocks[chosen_key])
+	alterer.update_mutant_bodyparts()
+
+/**
+ * Alter markings lets you add a particular body marking
+ */
+/datum/action/innate/alter_form/proc/alter_markings(mob/living/carbon/human/alterer)
+	var/list/candidates = GLOB.body_marking_sets
+	var/chosen_name = tgui_input_list(
+		alterer,
+		"Select which set of markings would you like to change into",
+		"Marking Alterations",
+		candidates,
+	)
+	if(!chosen_name)
+		return
+	var/datum/body_marking_set/marking_set = GLOB.body_marking_sets[chosen_name]
+	alterer.dna.species.body_markings = assemble_body_markings_from_set(marking_set, alterer.dna.features, alterer.dna.species)
+	alterer.update_body(is_creating = TRUE)
+
+/**
+ * Alter genitals lets you adjust the size or functionality of genitalia
+ * If you don't own the genital you try to adjust, it'll ask you if you want to add it first
+ */
+/datum/action/innate/alter_form/proc/alter_genitals(mob/living/carbon/human/alterer)
```

Comment:
There should be a check for genitals/body parts right off the bat. With the check, if it successfully finds genitals/body parts, it can add it to the tgui_input_list or something; would prevent a lot of checks for genitals/body parts later on.